lib/mayaUsd/render/MaterialXGenOgsXml/OgsFragment.cpp

```diff
@@ -223,7 +294,8 @@ std::string generateFragment(
     // MaterialX fragment).
     std::ostringstream nameStream;
     const size_t       sourceHash = std::hash<std::string> {}(fragmentSource);
-    nameStream << baseFragmentName << "__" << std::hex << sourceHash;
+    nameStream << baseFragmentName << "__" << std::hex << sourceHash
+               << OgsFragment::getSpecularEnvKey();
```

Comment:
Shader code depends on environment settings, so we must make sure we add the key everywhere we want to cache shaders.